CONTRIBUTING.md

```diff
@@ -0,0 +1,102 @@
+# Contributing to garage
+We welcome all contributions to garage.
+
+Use this guide to prepare your contribution.
+
+## Pull requests
+All contributions to the garage codebase are submitted via a GitHub pull request.
+
+### Review process
+To be submitted, a pull request must satisfy the following criteria:
+1. Rebases cleanly on the `master` branch
+1. Passes all continuous integration tests
+1. Conforms to the git commit message [format](#commit-message-format)
+1. Receives approval from another contributor
+1. Receives approval from a maintainer (distinct from the contributor review)
+
+These criteria may be satisfied in any order, but in practice your PR is unlikely to get attention from contributors until 1-3 are satisfied. Additionally, maintainer attention a scarce resource, so generally maintainers wait for a review from a non-maintainer contributor before reviewing your PR.
+
+## Code style
+The Python code in garage conforms to the [PEP8](https://www.python.org/dev/peps/pep-0008/) standard. Please read and understand it in detail.
+
+### garage-specific Python style
+These are garage-specific rules which are not part of the aforementioned style guides.
+* Python package imports should be sorted alphabetically within their PEP8 groupings. The sorting is alphabetical from left to right, ignoring case and Python keywords (i.e. `import`, `from`, `as`).
```

Comment:
Should we mention circular dependencies here as an exception?

Reply:
No, it's clear from the CI.

Reply:
Done